### ISSUE TITLE
Feature Request: JSX Error Boundary Component (or Try-Catch in JSX)

### ISSUE THREAD

**Issue by sjc5** [honojs/hono#1705]
### What is the feature you are proposing?

Hi there,

It would be great if we could catch errors and render fallbacks in JSX components, either by allowing normal try-catch syntax inside a JSX component, like this (which currently doesn't work):

```tsx
function ThisThrows() {
  throw new Error("Error");
  return <div>Won't be reached</div>;
}

function Fallback() {
  return <div>Caught -- Why won't this render?</div>;
}

function ThisShouldCatch() {
  try {
    return <ThisThrows />;
  } catch {
    return <Fallback />;
  }
}

app.get("/", async (c) => {
  // Results in 500 Internal Server Error, should it render Fallback?
  return c.html(<ThisShouldCatch />);
});
```

...or by introducing a new `ErrorBoundary` component, like this:

```tsx
import { ErrorBoundary } from "hono/jsx";

function ThisShouldCatch() {
  return (
    <ErrorBoundary fallback={<div>Caught</div>}>
      <ThisThrows />
    </ErrorBoundary>
  );
}
```

What do you think?

**Comment by usualoma** [comment]
Hi @sjc5!

Thanks for the suggestion.
I want to keep the spec in line with React and don't want to add too many "Hono-specific extensions."
On the other hand, I see the possibility of an exciting implementation of ErrorBoundary.
Please give me some time to think about it.

**Comment by yusukebe** [comment]
This is fixed with #1714. Now, you can use `ErrorBoundary`!
